src/msgraph_core/tasks/large_file_upload.py

```diff
@@ -143,7 +147,9 @@ def next_range(self):
     def next_range(self, value: Optional[str]) -> None:
         self._next_range = value
 
-    async def next_chunk(self, file: BytesIO, range_start: int = 0, range_end: int = 0) -> Future:
+    async def next_chunk(
+        self, file: BytesIO, range_start: int = 0, range_end: int = 0
+    ) -> LargeFileUploadSession:
```

Comment:
probably a novice question here: but shouldn't it be something like `Future[LargeFileUploadSession]` or does async also take care of that?

Reply:
Mypy specifies that coroutines (async def functions) should have return types annotated as the result type returned after awaiting another coroutine - https://mypy.readthedocs.io/en/latest/more_types.html#typing-async-await